sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -583,7 +583,9 @@ object FunctionRegistry {
           val validParametersCount = constructors
             .filter(_.getParameterTypes.forall(_ == classOf[Expression]))
             .map(_.getParameterCount).distinct.sorted
-          val expectedNumberOfParameters = if (validParametersCount.length == 1) {
+          val expectedNumberOfParameters = if (validParametersCount.isEmpty) {
+            constructors.headOption.map(_.getParameterCount).getOrElse(0).toString
+          } else if (validParametersCount.length == 1) {
```

Comment:
@yaooqinn, can you show the error message before and after this change, and fix the PR description and title? I think this PR now targets throw a better exception for the expression that has no constructors to call.